libraries/standard/mqtt/src/mqtt.c

```diff
@@ -146,9 +194,55 @@ MQTTStatus_t MQTT_Connect( MQTTContext_t * const pContext,
 
 MQTTStatus_t MQTT_Subscribe( MQTTContext_t * const pContext,
                              const MQTTSubscribeInfo_t * const pSubscriptionList,
-                             size_t subscriptionCount )
+                             size_t subscriptionCount,
+                             uint16_t packetId )
 {
-    return MQTTSuccess;
+    size_t remainingLength = 0UL, packetSize = 0UL, headerSize = 0UL;
+    int32_t bytesSent = 0;
+
+    /* Validate arguments. */
+    MQTTStatus_t status = validateSubscribeUnsubscribeParams( pContext,
+                                                              pSubscriptionList,
+                                                              subscriptionCount,
+                                                              packetId );
+
+    if( status == MQTTSuccess )
+    {
+        /* Get the remaining length and packet size.*/
+        status = MQTT_GetSubscribePacketSize( pSubscriptionList,
+                                              subscriptionCount,
+                                              &remainingLength,
+                                              &packetSize );
+        IotLogDebugWithArgs( "Subscribe packet size is %lu and remaining length is %lu",
+                             packetSize,
+                             remainingLength );
+    }
+
+    if( status == MQTTSuccess )
+    {
+        /* Serialize MQTT SUBSCRIBE packet. */
+        status = MQTT_SerializeSubscribe( pSubscriptionList,
+                                          subscriptionCount,
+                                          packetId,
+                                          remainingLength,
+                                          &( pContext->networkBuffer ) );
+    }
```

Comment:
Does MISRA require empty elses everywhere?

Reply:
Only if we have a if else-if blocks, MISRA 15.7 requires an else